src/groupeddataframe/splitapplycombine.jl

```diff
@@ -139,17 +139,17 @@ function groupby(df::AbstractDataFrame, cols;
     idxcols = index(df)[cols]
     intcols = idxcols isa Int ? [idxcols] : convert(Vector{Int}, idxcols)
     if isempty(intcols)
-        return GroupedDataFrame(df, intcols, ones(Int, nrow(df)),
-                                collect(axes(df, 1)), [1], [nrow(df)], 1, nothing,
-                                Threads.ReentrantLock())
+        return GroupedDataFrame(df, Symbol[], ones(Int, nrow(df)),
+                                nothing, nothing, nothing, nrow(df) == 0 ? 0 : 1,
```

Comment:
Because they can have `0` or `1` element (this was a bug to fill them before). Now we could conditionally fill them like we fill number of groups, but as filling them later is very cheap anyway I felt that setting them to `nothing` is OK.

Reply:
If computing the actual value here is trivial I'd do it, otherwise I agree it's cheap to compute later.

Reply:
I would leave it for later - this way code is more modular (otherwise we hardcode something here and can forget to update it if we change the default way to compute them in 5 years from now).